### DISCUSSION TITLE
How can I rank Users based on measure values?

### DISCUSSION THREAD

**Discussion by StefanDiric** [atoti/atoti#557] [Q&A] [answered]
Hello I'm new to atoti, 
I want **to rank my users by their profit and give them their rank**.
I have the measure `Profit.SUM` and hierarchy "User"  how can I proceed?

**Comment by GameTn** [comment, marked as answer]
Hi Stefan ,
I suggest you take a look at [`rank()`](https://docs.atoti.io/latest/lib/atoti/atoti.function.rank.html#atoti.rank).
What I understand from your problem is you want to rank users based on their profit?
Hope this work , if it doesn't please provide some more data because i don't know how your tables are defined also your structure.
`m["Rank"] = tt.rank(m["Profit.SUM"], h["User"], ascending=False)`
this is measure `Rank` it will do the job

**Reply by StefanDiric** [reply]
Yes it works Thanks
